pkgs/tools/filesystems/bcachefs-tools/default.nix

```diff
@@ -2,77 +2,94 @@
 , stdenv
 , fetchFromGitHub
 , pkg-config
-, docutils
 , libuuid
-, libscrypt
 , libsodium
 , keyutils
 , liburcu
 , zlib
 , libaio
 , zstd
 , lz4
-, python3Packages
-, util-linux
+, attr
 , udev
 , valgrind
 , nixosTests
-, makeWrapper
-, getopt
 , fuse3
+, cargo
+, rustc
+, coreutils
+, rustPlatform
+, makeWrapper
 , fuseSupport ? false
 }:
-
-stdenv.mkDerivation {
+let
+  rev = "5ef62f56ab50c5799f713e3a42f5c7ad7e8283d3";
+in stdenv.mkDerivation {
   pname = "bcachefs-tools";
-  version = "unstable-2023-01-31";
+  version = "unstable-2023-05-13";
 
   src = fetchFromGitHub {
     owner = "koverstreet";
     repo = "bcachefs-tools";
-    rev = "3c39b422acd3346321185be0ce263809e2a9a23f";
-    hash = "sha256-2ci/m4JfodLiPoWfP+QCEqlk0k48zq3mKb8Pdrtln0o=";
+    inherit rev;
+    hash = "sha256-w8ez7+E4suPN0W5OzbzFLysDXlmZrjFk20f8VZlIfuE=";
   };
 
-  postPatch = ''
-    patchShebangs .
-    substituteInPlace Makefile \
-      --replace "pytest-3" "pytest --verbose" \
-      --replace "INITRAMFS_DIR=/etc/initramfs-tools" \
-                "INITRAMFS_DIR=${placeholder "out"}/etc/initramfs-tools"
-  '';
-
   nativeBuildInputs = [
-    pkg-config docutils python3Packages.python makeWrapper
+    pkg-config
+    cargo
+    rustc
+    rustPlatform.cargoSetupHook
+    rustPlatform.bindgenHook
+    makeWrapper
   ];
 
+  cargoRoot = "rust-src";
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "bindgen-0.64.0" = "sha256-GNG8as33HLRYJGYe0nw6qBzq86aHiGonyynEM7gaEE4=";
+    };
+  };
```

Comment:
Mind explaining to me what is the advantage of embedding the `Cargo.lock` here? Instead of using from `"${src}/${cargoRoot}/Cargo.lock";`.

The only things I see are cons:
- Evaluation time goes up;
- Nixpkgs objects when cloning are bigger;
- Reviewing has a bigger diff to cover;
- Manually bumping the version now means you also have to bump the `./Cargo.lock`;
- When overriding, you'll need to override `cargoDeps` besides `src` and `version`;
- Using it from `src` already guarantees reproducibility since we have the source hashed.

Reply:
importCargoLock reads the Cargo.lock file, which is an import-from-derivation which AIUI is not allowed in nixpkgs